### PR TITLE
Use Docc for documentation

### PR DESCRIPTION
**Pull request by Lukasa**
Motivation

Documentation is nice, and we can help support users by providing useful
clear docs.

Modifications

Add Docc to 5.6 and later builds
Make sure symbol references work
Add overview docs

Result

Nice rendering docs

### CODE REVIEW

**Comment by Lukasa** [comment]
Uh, this is new. @yim-lee any idea what's happened to SwiftPM here? It seems to have generated a test file that emits a compiler warning, which we choke on because we use warnings as errors.

File: Sources/NIOTransportServices/NIOTSListenerBootstrap.swift [hunk 1]
```diff
@@ -175,7 +175,7 @@ public final class NIOTSListenerBootstrap {
 
     /// Specifies a `ChannelOption` to be applied to the `NIOTSListenerChannel`.
     ///
-    /// - note: To specify options for the accepted `NIOTSConnectionChannels`s, look at `NIOTSListenerBootstrap.childChannelOption`.
+    /// > Note: To specify options for the accepted `NIOTSConnectionChannel`s, look at ``childChannelOption(_:value:)``.
```
Comment (FranzBusch) [inline Sources/NIOTransportServices/NIOTSListenerBootstrap.swift:178]: Just out of curiosity is `- Note` not working in Docc anymore?

**Review by FranzBusch — APPROVED** [review]
LGTM!

**Comment by Lukasa** [comment]
Oh dang it, I know what's happening: this is because there _are no tests_ on Linux. Ok, I'll add a dummy test.

File: Sources/NIOTransportServices/NIOTSListenerBootstrap.swift (hunk 1 above)
Comment (Lukasa) [reply Sources/NIOTransportServices/NIOTSListenerBootstrap.swift:178]: I dunno, they encourage use of this angle bracket version in the docs so I did that.

**Comment by yim-lee** [comment]
🎉

**Pull request merged**